drake/solvers/OptimalMotionPlanningTree.m

```diff
@@ -0,0 +1,342 @@
+  %adds the following properties
+%     C             cost array
+%     gamma         RRT* parameter
+%     steerFactor   radius for the steering function
+%     traj          indices of the optimal trajectory
+%     eta           radius for tree rewiring
+%     costType      type of cost
+%     xGoal         
+
+  properties
+    C
+    gamma
+    steerFactor
+    traj
+    eta
+    costType
+    xGoal
+  end
+  
+  methods
+    
+    function obj = OptimalMotionPlanningTree(robot, endEffectorId, endEffectorPoint)
+      obj = obj@TaskSpaceMotionPlanningTree(robot, endEffectorId, endEffectorPoint);
+    end
+    
+    function [obj, id_last] = init(obj, q_init)
+      obj.C = NaN(1, obj.N);
+      obj.C(1) = 0; %An initial cost might be needed if not using path length
+      obj.eta = obj.steerFactor;
+      
+      %Compute Gamma
+      R3Tree = obj.trees{obj.tspace_idx}.trees{1};
+      freeSpaceVolume = prod(R3Tree.sampling_ub - R3Tree.sampling_lb);
+      unitBallVolume = 16/105*pi^3;
+      d = 7; %for 7D space
+      obj.gamma = (2*(1+1/d))^(1/d)*(freeSpaceVolume/unitBallVolume)^(1/d);
+      
+      [obj, id_last] = init@TaskSpaceMotionPlanningTree(obj, q_init);
+    end
+    
+    function  [obj, id] = addVertex(obj, q, id_parent, cost)
+      [obj, id] = addVertex@TaskSpaceMotionPlanningTree(obj, q, id_parent);
+      if nargin < 4
+        obj.C(id) = obj.C(id_parent) + obj.cost(id, id_parent);
+      else
+        obj.C(id) = cost;
+      end
+    end
+    
+    function qPath = rebuildTraj(obj, x_goal)      
+      if all(obj.getVertex(obj.traj(1)) == x_goal)
+        obj.traj = fliplr(obj.traj);
+      end
+      qPath = [];
+      for i = 2:length(obj.traj)
+        qPath =  [qPath obj.getVertex(obj.traj(i-1))]; %#ok<AGROW>
+        cSpaceTree = obj.trees{obj.cspace_idx};
+        qNew = cSpaceTree.getVertex(obj.traj(i-1));
+        points = obj.getVertex([obj.traj(i), obj.traj(i-1)]);
+        d = obj.trees{obj.tspace_idx}.distanceMetric(points(1:7, 1), points(1:7, 2));
+        n = ceil(d/obj.max_edge_length);
+        edgePath = NaN(obj.num_vars, n-1);
+        for j = 1:n-1
+          pt = obj.interpolate(points(:,2), points(:,1), j/n);
+          constraints = obj.generateEndEffectorConstraints(pt(1:7));
+          qNew = cSpaceTree.solveIK(pt(8:end), qNew, constraints);
+          edgePath(:,j) = [pt(1:7); qNew];
+        end
+        qPath = [qPath edgePath];
+      end
+      qPath = [qPath x_goal];
+    end
+    
+    function obj = addPointToTraj(obj, point, p1, p2)      
+      if obj.getParentId(obj.traj(p2)) == obj.traj(p1)
+        obj = obj.addVertex(point, obj.traj(p1));
+        obj = obj.setParentId(obj.traj(p2), obj.n);
+      else
+        obj = obj.addVertex(point, obj.traj(p2));
+        obj = obj.setParentId(obj.traj(p1), obj.n);
+      end
+      obj = obj.getTrajToGoal();
+    end
+    
+    function obj = rrtStarIteration(obj, status, iter, biasSpacing)
```

Comment:
Would it be a lot of work to add an `rrtStar` method? It'd be nice to be able to use this class outside of the context of `MultipleTreeProblem`.